### PR TITLE
Align create_asset_folder() logic to original intent

### PR DESCRIPTION
**Pull request by mattskone**
Fixes https://asudev.jira.com/browse/ACA-1555

BREAKING CHANGE

Unrelated to this bug, I also updated two package dependencies due to dependabot alerts.

### CODE REVIEW

File: geeutils/eu.py [hunk 1]
```diff
@@ -62,30 +62,34 @@ def authenticate(allow_interactive=True) -> None:
...
 
     Also create intermediate folders along the path, if they don't already
     exist.
 
     Args:
-        asset_id: A path to the folder to be created.
+        folder_path: A path to the folder to be created. Note that GEE expects the path
+            to start with:
+            users/[user_name]/, or
+            projects/[project_name]/
```
Comment (mattskone) [inline geeutils/eu.py:75]: Important: change the docstring

File: geeutils/eu.py [hunk 2]
```diff
@@ -62,30 +62,34 @@ def authenticate(allow_interactive=True) -> None:
...
         raise ValueError(
-            "GEE asset id {} is too short (must have more than 2 path segments)".format(
-                asset_id
+            (
+                f"GEE folder path {folder_path} is too short "
+                "(must have more than 2 path segments)"
             )
         )
-    for i in range(2, len(path_segments) - 1):
+    for i in range(2, len(path_segments)):
```
Comment (mattskone) [inline geeutils/eu.py:92]: Important: change the logic

**Review by eldang — APPROVED** [review]
Looks sensible to me.  Do we know of other code which this change will affect, apart from what you're currently working on wrt the bleaching monitoring pipeline?

**Comment by mattskone** [comment]
Among active Atlas repos, the only other invocation of this method occurs during turbidity generation, but:
1. It's not a turbidity option that we normally use, and
2. The turbidity code is pegged to the previous (`1.1.0`) version of this package, so it won't be an issue unless/until we update the turbidity `Pipfile`.

All other invocations occur in archived repos.

**Comment by eldang** [comment]
Ah, good point!  The version pegging will avoid the unintended consequences I was worrying about.

**Comment by mattskone** [comment]
Exactly - this package (and `proc_gcs_utils`) are public, distributed via PyPI, and strictly pegged to version numbers, so we don't face the same vulnerability as when we use `latest` Docker images.

**Pull request merged**